components/raftstore/src/coprocessor/mod.rs

```diff
@@ -278,12 +304,10 @@ impl CmdBatch {
 }
 
 pub trait CmdObserver<E>: Coprocessor {
-    /// Hook to call after preparing for applying write requests.
-    fn on_prepare_for_apply(&self, cdc_id: &ObserveHandle, rts_id: &ObserveHandle, region_id: u64);
-    /// Hook to call after applying a write request.
-    fn on_apply_cmd(&self, cdc_id: ObserveID, rts_id: ObserveID, region_id: u64, cmd: &Cmd);
     /// Hook to call after flushing writes to db.
-    fn on_flush_apply(&self, engine: E);
+    fn on_flush_applied_cmd_batch(&self, cmd_batches: Vec<CmdBatch>, engine: E);
```

Comment:
Why not just `&mut Vec<CmdBatch>`? So only one method is necessary.

Reply:
>  it depends on the observer's priority, 

That's the exact design goal of `priority`. Using different methods is too verbose.